tabby-electron/src/shells/wsl.ts

```diff
@@ -58,7 +68,24 @@ export class WSLShellProvider extends ShellProvider {
 
         if (lxss?.DefaultDistribution) {
             const defaultDistKey = wnr.getRegistryKey(wnr.HK.CU, lxssPath + '\\' + String(lxss.DefaultDistribution.value))
+            let _icon = wslIconMap.Linux
             if (defaultDistKey?.DistributionName) {
+                // check if the register has PackageFamilyName
+                if (defaultDistKey.PackageFamilyName) {
+                    // get the icon from the package family name
+                    const packageFamilyName = (defaultDistKey.PackageFamilyName.value as string).split('_')[0]
+
+                    if (packageFamilyName) {
```

Comment:
Extract the common steps into a function here

Reply:
Done